README.md

```diff
@@ -49,10 +49,147 @@ $ yarn lint
+  reporters: ['default', '../dist/reporter.js']
+};
+```
+
+### Test with mocked traffic
+
+```shell
+$ yarn test
+```
+
+This command runs tests with mocked traffic and therefore no live traffic is enabled. You can find recordings that this command uses as mocked traffic in `/nock`.
+
+Recordings are categorised similarly as integrations in `/grid` folder:
+`nock/<scope>/<profile-name>/<provider-name>/<usecase-name>/recording-<hash>.json`
+
+- every recording gets hashed based on **input**, **test name** or **custom test name**
+- this hash can be only seen with use of `DEBUG=superface:testing:hash*`
+
+### Test with live traffic
+
+```shell
+$ yarn test:record
+```
+
+With this command, live traffic is enabled and therefore maps send requests to the live providers. This command also records this request and response. There are multiple scenarios what can happen:
+
+- use-case does not have recording → records traffic and saves as default one
+- use-case already have recording → records new traffic and compares it to the old one →
+    - if it matches → do not save recording
+    - if it doesn’t match → save new recording next to old one with suffix `-new`
+
+### Test with new mocked traffic
+
+```shell
+$ yarn test:with_new_traffic
+```
+
+This command run tests without live traffic, similar as `yarn test`, but uses new recordings instead of default ones (if present). New recordings are always newly recorded traffic that does not match the default old one and are located next to the old one with suffix `-new`.
+
+### Test and replace old recording with new one
+
+**⚠️ This command should be run only when you’re sure that tests are passing with new recordings → `yarn test:with_new_traffic`**
+
+```shell
+$ yarn test:update_traffic
+```
+
+This commands run tests with mocked traffic, but also replaces old recording with new one (if present) before loading recordings - this means that tests are run with mocked new traffic already.
+
+### Test with live traffic with development reporting
+
+```shell
+$ yarn test:record:dev
+```
+
+This command run tests with live traffic and record it, similar as `yarn test:record`. The script implementation looks like this:
+
+```shell
+$ TEST_ENV='dev' yarn test:record --group=live/safe
+```
+
+`TEST_ENV` sets development environment for reporter - to use local reporter that just logs the provider changes into console
+
+`--group=live/safe` filters what tests to run, thanks to [jest-runner-groups](https://www.npmjs.com/package/jest-runner-groups) 
+
+### Test with live traffic with production reporting
+
+```shell
+$ yarn test:record:prod
+```
+
+This command run tests with live traffic and record it, similar as `yarn test:record` or `yarn test:record:dev`. Only difference between `test:record:dev` is that it uses production reporter that reports provider changes to the public slack channel.
+
+Slack channel can be changed with env variable `PROD_REPORTING_DESTINATION`. It expects channel id.
```

Comment:
Ooof, suddenly you drop Slack reporting on me. Like you tell me how to change the Slack channel ID without any prior info on how to setup the reporter, Slack bot, access tokens etc. I don't think this should be documented here, but in a separate document, but even then it will be probably only for us.

```suggestion
Run tests under `live/safe` group with live traffic, record it and report provider changes to Slack.

Set Slack channel ID with env variable `PROD_REPORTING_DESTINATION`.
```

Reply:
Good point, some section about setting up reporter would be good.